### PR TITLE
test(widget): add unit tests for Paragraph

### PR DESCRIPTION
**Pull request by joshka**
Putting this in place to help ensure #149 doesn't break anything.

I wasn't 100% certain how the `zero_width_char_at_end_of_line` test was supposed to actually fail / not-fail as there's no assertion. Rewrote it to conform to the pattern of the other tests (but perhaps @orhun and @rhysd can add a bit more context if this is incorrect?

### CODE REVIEW

**Comment by joshka** [comment]
Ping @TimerErTim for a review

**Comment by joshka** [comment]
Force pushed formatting fixes

**Review by TimerErTim — COMMENTED** [review]
The test cases generally look good, I will integrate them into my branch. I will extend them there when I can think of new/expanded ones.

File: src/widgets/paragraph.rs [hunk 1]
```diff
@@ -202,13 +202,220 @@ impl<'a> Widget for Paragraph<'a> {
 
 #[cfg(test)]
 mod test {
+    use crate::{
+        style::Color,
+        text::{Span, Spans},
+        widgets::Borders,
+    };
+
```
Comment (orhun) [inline src/widgets/paragraph.rs:210]: nit: can we import `use super::*;` before this line to make the formatting consistent with other modules?

Comment (joshka) [reply src/widgets/paragraph.rs]: Done

**Comment by joshka** [comment]
Rebased on #161 to use `assert_buffer_eq!` instead of `assert_eq`

**Review by orhun — APPROVED** [review]

**Pull request merged**